lib/eclipse/EclipseState/Schedule/MSW/Compsegs.cpp

```diff
@@ -168,18 +168,61 @@ namespace Opm {
             }
 
             compseg.m_segment_number = segment_number;
+
+            // when depth is default or zero, we obtain the depth of the completion based on the information
+            // of the related segments
             if (compseg.m_center_depth == 0.) {
-                // using the depth of the segment node as the depth of the completion
-                // TODO: now only one completion for one segment is hanlded,
-                // TODO: later we will try to handle more than one completion for each segment,
-                // TODO: which will be a linear interpolation based on the segment node depth
-                // TODO: in the same branch, while the actually way is not clear yet
-                const int segment_location = segment_set.numberToLocation(segment_number);
-                compseg.m_center_depth = segment_set[segment_location].depth();
+                compseg.calculateCenterDepthWithSegments(segment_set);
             }
         }
     }
 
+    void Compsegs::calculateCenterDepthWithSegments(const SegmentSet& segment_set) {
+
+        // the depth and distance of the segment to the well head
+        const int segment_location = segment_set.numberToLocation(m_segment_number);
```

Comment:
Instead of using variable `segment_location`[1] I think this would be easier to read if it were something like this:
```C++
const auto& segment = segment_set.getFromNumber( m_segment_number );
// And then continue with the segment reference.
```



[1]: `segment_location` is an integer and should probably be named something like `segment_index` or `segment_nr`? The name `segment_location`hints at something with dimension length?

Reply:
the naming is hard here. `segment number` is a ID, can be any positive number and does not mean anything about the position of the segment in the whole well.  `segment_nr` I am worrying can be a little confusing with `segment number`. 

Basically, `segment_location` means the location of the segment in the whole vector of segments. To me, `segment_location` and `segment_index` are equally good / bad. `well index` in well can be something related to transmissibility / injectvity (all different though).  Anyway, I will change all the location to index, and add some comments there.

Reply:
Well - to me `segment_index` - or just `index` is much better.